src/Utils.cc

```diff
@@ -652,17 +652,17 @@ void radix_sort_kernel(
     const V* const input_values,
     K* const output_keys,
     V* const output_values,
-    const int elements_count,
-    int* const histogram,
-    int* const histogram_ps,
+    const int64_t elements_count,
+    int64_t* const histogram,
+    int64_t* const histogram_ps,
     const int pass,
     const bool pass_with_sign_bit = false) {
   const auto tid = omp_get_thread_num();
   const auto nthreads = omp_get_num_threads();
   const auto elements_count_4 = elements_count / 4 * 4;
 
-  int* const local_histogram = &histogram[RDX_HIST_SIZE * tid];
-  int* const local_histogram_ps = &histogram_ps[RDX_HIST_SIZE * tid];
+  const auto local_histogram = &histogram[RDX_HIST_SIZE * tid];
+  const auto local_histogram_ps = &histogram_ps[RDX_HIST_SIZE * tid];
```

Comment:
```suggestion
  const auto* const local_histogram = &histogram[RDX_HIST_SIZE * tid];
  const auto* const local_histogram_ps = &histogram_ps[RDX_HIST_SIZE * tid];
```

Reply:
@r-barnes You probably meant here `auto* const` instead of `const auto* const`, which indeed is a better option than `const auto`. Code changed.